### PR TITLE
Various enhancements in configuration

### PR DESCRIPTION
**Pull request by qwerty287**
- backends: move to cli flags instead of os.Getenv
- ssh: support 2fa with key and password
- allow to set grpc jwt secret (solves todo)
- allow to set default and max timeout (solves todo)

Closes https://github.com/woodpecker-ci/woodpecker/issues/896
Closes https://github.com/woodpecker-ci/woodpecker/issues/1131

### CODE REVIEW

File: cmd/agent/flags.go [hunk 1]
```diff
@@ -108,7 +108,50 @@ var flags = []cli.Flag{
 		Value:   "auto-detect",
 	},
 
-	// TODO: add flags of backends
+	// backend docker
```
Comment (lafriks) [inline cmd/agent/flags.go:111]: can these flags be moved to engine (function in interface that would return them)? And than append them by looping all engines and calling that func?

Comment (6543) [reply cmd/agent/flags.go:111]: I had a similar idea what if we could move backend flag declaration into pipeline/backend/x/flags.go ...

but if we want to have a simple codebase for now we have to do it as is ... and I might come up with an elegant idea later :)

File: cmd/server/flags.go [hunk 2]
```diff
@@ -120,6 +126,18 @@ var flags = []cli.Flag{
 		Usage:   "The default docker image to be used when cloning the repo",
 		Value:   constant.DefaultCloneImage,
 	},
+	&cli.Int64Flag{
+		EnvVars: []string{"WOODPECKER_DEFAULT_PIPELINE_TIMEOUT"},
+		Name:    "default-pipeline-timeout",
+		Usage:   "The default time for a repo in minutes before a pipeline gets killed",
```
Comment (6543) [inline cmd/server/flags.go:132]: minutes seconds ...?

Comment (6543) [inline cmd/server/flags.go:132]:
```suggestion
		Usage:   "The default time in minutes for a repo in minutes before a pipeline gets killed",
```

File: pipeline/backend/docker/docker.go [hunk 3]
```diff
@@ -41,9 +41,6 @@ type docker struct {
 	volumes    []string
 }
 
-// make sure docker implements Engine
```
Comment (6543) [inline pipeline/backend/docker/docker.go:44]: why rm?

File: pipeline/backend/local/local.go [hunk 4]
```diff
@@ -47,9 +47,6 @@ type local struct {
 	workingdir string
 }
 
-// make sure local implements Engine
```
Comment (6543) [inline pipeline/backend/local/local.go:50]: why rm?

File: pipeline/backend/ssh/ssh.go [hunk 5]
```diff
@@ -29,9 +28,6 @@ func (c readCloser) Close() error {
 	return nil
 }
 
-// make sure local implements Engine
```
Comment (6543) [inline pipeline/backend/ssh/ssh.go:32]: same

File: cmd/agent/flags.go (hunk 1 above)
Comment (anbraten) [reply cmd/agent/flags.go:111]: Do we need to know which backend is active / selected to get those flags. Couldn't we just get a list of all available options from each backend and add them to the list.

Comment (6543) [reply cmd/agent/flags.go:111]: if we just import things ...
we get:
 - import circles and
 - https://woodpecker-ci.org/docs/next/development/architecture get violated

**Review by 6543 — APPROVED** [review]

**Pull request merged**